pytorch_lightning/utilities/distributed.py

```diff
@@ -376,3 +376,30 @@ def init_dist_connection(
             f"All distributed processes registered. Starting with {world_size} processes\n"
             f"{'-' * 100}\n"
         )
+
+
+def _collect_states_on_rank_zero(state: Dict[str, Any], device: torch.device) -> Optional[Dict[int, Any]]:
+    """This distributed utility collects dictionary state across all processes.
+
+    Args:
+        state: Dictionary contain the state of the current process
+        device: Current process device.
+
+    Returns:
+        states: On global rank 0, a dictionary where the primary keys are
+            the process rank and the values their associated states. Otherwise, returns None.
+    """
+    if not distributed_available():
+        return {0: state}
+    states = {}
+    for rank in range(1, torch.distributed.get_world_size()):
+        if torch.distributed.get_rank() == rank:
+            objects = [state]
+        else:
+            objects = [None]  # type: ignore
+        torch.distributed.broadcast_object_list(objects, src=rank, device=device)
+        states[rank] = objects[0]
+    if torch.distributed.get_rank() != 0:
```

Comment:
nit: there are two calls to `get_rank` in this function, call it once and save the value instead

Reply:
Done !